src/plugins/irc/irc-server.c

```diff
@@ -4032,6 +4042,51 @@ irc_server_gnutls_callback (void *data, gnutls_session_t tls_session,
         fingerprint = IRC_SERVER_OPTION_STRING (server,
                                                 IRC_SERVER_OPTION_SSL_FINGERPRINT);
 
+        /* get ca_file option in server */
+        ca_file0 = (char *) IRC_SERVER_OPTION_STRING (server,
+                                                IRC_SERVER_OPTION_SSL_CA_FILE);
+
+        if (ca_file0 && ca_file0[0])
+        {
+            weechat_dir = weechat_info_get ("weechat_dir", "");
+            ca_file1 = weechat_string_replace (ca_file0, "%h", weechat_dir);
+            ca_file2 = (ca_file1) ?
+                weechat_string_expand_home (ca_file1) : NULL;
+
+            if (ca_file2)
+            {
+                weechat_printf (
+                    server->buffer,
+                    _("%sgnutls: set ca_file to %s"),
+                    weechat_prefix ("network"),
+                    ca_file2);
+                gnutls_certificate_allocate_credentials (&server->gnutls_xcred);
+                gnutls_certificate_set_x509_trust_file (server->gnutls_xcred, ca_file2,
+                                                                GNUTLS_X509_FMT_PEM);
+                gnutls_credentials_set(tls_session, GNUTLS_CRD_CERTIFICATE, server->gnutls_xcred);
+
+                /*
+                 * We need to set verify/retrieve functions for our server
+                 * specific gnutls_xcred here.
+                 */
+                #if LIBGNUTLS_VERSION_NUMBER >= 0x02090a /* 2.9.10 */
```

Comment:
Preprocessor statements shouldn't be indented, regardless of their nested position.